.eslintrc.js

```diff
@@ -57,6 +64,16 @@ module.exports = {
       env: {
         embertest: true
       }
+    },
+
+    // all TypeScript files
+    {
+      files: ['**/*.ts'],
+      rules: {
+        // These are covered by tsc
+        'no-undef': 'off',
+        'no-unused-vars': 'off'
+      }
```

Comment:
This is no critique against this PR in any way and more of a note to myself and other maintainers: I personally have a great distaste for these `overrides`-based eslint configs.

At our company we use multiple `root` eslint configs per package with great success. This works especially well, when you have browser TypeScript and Node JavaScript files in the same pacakge.

We might want to use a setup like the one suggested in [`@clark/eslint-config-ember`](https://github.com/ClarkSource/eslint-config/tree/master/packages/eslint-config-ember) in the future.